administration/src/application/components/forms/RegionForm.tsx

```diff
@@ -0,0 +1,69 @@
+import { Region, useGetRegionByPostalCodeQuery } from '../../../generated/graphql'
+import { useUpdateStateCallback } from '../../useUpdateStateCallback'
+import { Form } from '../../FormType'
+import { CompoundState, createCompoundGetArrayBufferKeys, createCompoundInitialState } from '../../compoundFormUtils'
+import SelectForm, { SelectItem } from '../primitive-inputs/SelectForm'
+import { ProjectConfigContext } from '../../../project-configs/ProjectConfigContext'
+import { useContext } from 'react'
+import { useSnackbar } from 'notistack'
+
+const SubForms = {
+  region: SelectForm,
+}
+
+type State = CompoundState<typeof SubForms>
+type ValidatedInput = { regionId: number }
+type Options = { regions: Region[] }
+type AdditionalProps = { postalCode: string }
+
+const getOptionsLabel = (prefix: string, name: string) => `${name} (${prefix})`
+
+export const getOptions = (regions: Region[]): SelectItem[] =>
+  regions.map(region => {
+    return { label: getOptionsLabel(region.prefix, region.name), value: region.id.toString() }
+  })
+
+const RegionForm: Form<State, Options, ValidatedInput, AdditionalProps> = {
+  initialState: { ...createCompoundInitialState(SubForms) },
+  getArrayBufferKeys: createCompoundGetArrayBufferKeys(SubForms),
+  validate: (state, options) => {
+    const result = SubForms.region.validate(state.region, { items: getOptions(options.regions) })
+    if (result.type === 'error') {
+      return { type: 'error' }
+    }
+    return { type: 'valid', value: { regionId: Number(result.value.shortText) } }
+  },
+  Component: ({ state, setState, options, postalCode }) => {
+    const projectId = useContext(ProjectConfigContext).projectId
+    const { enqueueSnackbar } = useSnackbar()
+    const {} = useGetRegionByPostalCodeQuery({
+      onError: () => {
+        enqueueSnackbar('Region konnte nicht ermittelt werden. Bitte wählen sie ihre Region manuell aus.', {
+          variant: 'warning',
+        })
+      },
+      onCompleted: result => {
+        if (result.region) {
+          const { name, prefix, id } = result.region
+          enqueueSnackbar(`${getOptionsLabel(prefix, name)} wurde als ihre Region ermittelt.`, {
+            variant: 'success',
+          })
+          setState(() => ({ region: { selectedValue: id.toString() } }))
+        }
+      },
+      variables: { postalCode: postalCode, projectId: projectId },
+      skip: postalCode.length !== 5,
+    })
+
+    return (
+      <SubForms.region.Component
```

Comment:
Add a `<Typography>` tag with a text with something like:

In der Regel müssen Sie die Ehrenamtskarte in dem Landkreis oder der kreisfreien Stadt Ihres Hauptwohnsitzes beantragen. Weitere Informationen können Sie [hier einsehen](https://www.lbe.bayern.de/engagement-anerkennen/ehrenamtskarte/voraussetzungen/index.php).
Eine Liste der teilnehmenden Landkreise und kreisfreien Städte finden Sie unten.

And below add an alert (see https://mui.com/material-ui/react-alert/) with something like


{If erfolgreich} `<Alert sucess>`Die zuständige Behörde konnte anhand Ihrer Postleitzahl automatisch ermittelt werden.`</Alert>`
{If keine gültige PLZ} `<Alert info>`Bitte geben Sie oben eine 5-stellige Postleitzahl an, sodass die zuständige Behörde automatisch ermittelt werden kann.`</Alert>`
{If loading} `<CircularProgress />`
{Else:} `<Alert warn>`Die zuständige Behörde konnte nicht automatisch anhand Ihrer Postleitzahl ermittelt werden.`</Alert>`